src/dfcx_scrapi/tools_ml/utterance_generator_utils.py

```diff
@@ -0,0 +1,369 @@
+                full path to agent
+            intent_subset: display names of the intents to create a test
+                for, base phrases come from the training in the intent.
+            dataset_size: overall target size of the test set to create, may
+                be less depending if new independent phrases can be generated
+                from the data. The function tries to get even entries per
+                intent.
+
+        Returns:
+            Dataframe with columns:
+                utterance: synthesized phrases.
+                intent: intent the utterance was generated from, 
+                    also true label.
+        """
+        synthetic_dataset = self.create_synthetic_dataset(agent_id, intent_subset, 
+            dataset_size)
+        test_dataset = (
+            synthetic_dataset.copy()[["synthetic_phrases", "intent"]]
+            .rename(columns={"synthetic_phrases": "utterance"})
+            .reset_index(drop=True)
+        )
+        return test_dataset
+
+    def create_new_training_phrases(
+        self, 
+        agent_id: str, 
+        intent_subset: List[str], 
+        new_phrases: int = 100
+    ) -> pd.DataFrame:
+        """ Creates new training phrases for a given list of intents. 
+        
+        The phrases in this set will not be exact string match phrases which
+        exist in the training phrases but will be close semantically. This can
+        be used to run through the core.intents modify_training_phrase_df
+        function to create a new training phrase dataframe. This and a
+        parameters dataframe can be run through tools.dataframe_functions
+        bulk_update_intents_from_dataframe function to make the updates in a
+        dialogflow agent. The new_training output dataframe can be used as the
+        input actions dataframe in the bulk_update_intents_from_dataframe
+        function.
+
+        Args:
+            agent_id: name parameter of the agent to pull intents from - full
+                path to agent.
+            intent_subset: display names of the intents to create a new phrases
+                for, base phrases come from the training in the intent.
+            new_phrases: overall target size of new phrases to create, may be
+                less depending if new independent phrases can be generated from
+                the data. The function tries to get even entries per intent.
+
+        Returns:
+            Dataframe with columns:
+                display_name: intent to add the phrase to.
+                phrase: new phrase. 
+                action: "add".
+        """
+        synthetic_dataset = self.create_synthetic_dataset(
+            agent_id, intent_subset, new_phrases
+        )
+        new_training = (
+            synthetic_dataset.copy()[["intent", "synthetic_phrases"]]
+            .rename(columns={
+                "intent": "display_name", "synthetic_phrases": "phrase"})
```

Comment:
@Hgithubacct  Can we keep the intent column's name as "intent" and remove the rename to "display_name"? Just for consistent namespace.

Reply:
@Greenford Ack. I've made this change.